### PR TITLE
Feature/sandhya hub

### PR DESCRIPTION
**Pull request by Ragingblues**
Create Initiative form files committed. This is not a full fledged code yet. There are some issues with the dropdown in the form. Working on trying to fix it.

### CODE REVIEW

**Comment by jc2016** [comment]
COuld you name the feature branch according to the name of the feature you are working on
cc: @ragasudhaa

File: packages/ui/src/components/createInitiativeForm.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          floatingLabelText="Approver Name"          
+          name="approverName"
+          onChange={onChange}
+          value={user.approverName}
+          errorText={errors.approverName}
+        />
+      </div>
+
+      <div className="button-line">
+        <RaisedButton type="submit" label="Log in" primary />
```
Comment (shalinijha92) [inline packages/ui/src/components/createInitiativeForm.js]: Button says "Login" - Change it to "Create Initiative"

File: packages/ui/src/components/createInitiativeForm.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        />
+      </div>
+      <div>
+        <SelectField
+            floatingLabelText="Participants"
+            value={totalParticipants}
+            onChange={(event, index, value) => onParticipantDropDownChange(event, index, value)}
+            autoWidth={true}
+          >
+            <MenuItem value={1} primaryText="Never" />
```
Comment (shalinijha92) [inline packages/ui/src/components/createInitiativeForm.js]: On selecting an option from the dropdown, it doesn't show up in the UI area.
A better idea here would be to create a constant with various options and iterate through the constant and render the MenuItem, cause options are gonna be dynamic and not hard coded

File: packages/ui/src/components/createInitiativeForm.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  errors,
+  user,
+  pastDateCheck,
+  currentDate= new Date(),
+  totalParticipants,
+  onParticipantDropDownChange
+
+}) => (
+  <Card className="container create-initiative-page">
+  <form onSubmit={onSubmit}>
```
Comment (shalinijha92) [inline packages/ui/src/components/createInitiativeForm.js]: Nothing happens onSubmit, it just consoles the username and password. Instead you should make an API call, if present, else store the form data in redux.

File: packages/ui/src/components/createInitiative.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    }
+    handleValidation() {
+        let fields = this.state.user;
+        let errors = {};
+        let formIsValid = true;
+
+        if (!fields["initiativeName"]) {
+            formIsValid = false;
+            errors["initiativeName"] = "Enter Initiative Name";
```
Comment (shalinijha92) [inline packages/ui/src/components/createInitiative.js]: This messages should come from a constants, create a message.js file in constants and import all the constants from there

File: packages/ui/src/components/createInitiative.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        this.setState({
+            totalParticipants:value
+        });
+        console.log(this.state.totalParticipants);
+    };
+
+    /**
+     * Render the component.
+     */
+    render() {
```
Comment (shalinijha92) [inline packages/ui/src/components/createInitiative.js]: Follow ES6 fat arrow syntax everywhere, it should be consistent

File: packages/ui/src/components/createInitiative.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        return ( <
+            CreateInitiativeForm onSubmit = {
```
Comment (shalinijha92) [inline packages/ui/src/components/createInitiative.js]: Remove this and bring the entire Form component here.

File: packages/ui/src/components/createInitiative.js [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+        this.state = {
+            errors: {},
+            user: {
+                initiativeName: '',
+                approverName: ''
+            },
+            totalParticipants: {}
+        };
+
+        this.processForm = this.processForm.bind(this);
```
Comment (shalinijha92) [inline packages/ui/src/components/createInitiative.js]: .bind() is not needed when using ES6 fat arrow functions

File: packages/ui/src/css/app.css [hunk 8]
```diff
@@ -21,6 +21,15 @@
 .App-intro {
   font-size: large;
 }
+.create-initiative-page{
+  padding: 20px 50px;
+  width: 600px;
+  margin: 0 auto;
+  text-align: center;
+}
+.create-initiative-page .field-line{
```
Comment (shalinijha92) [inline packages/ui/src/css/app.css]: Remove this if not needed

**Pull request closed without merging**